testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -1615,4 +1698,272 @@ mod test {
             .expect("Unable to deserialize array of MessageSignature");
         assert_eq!(event_signer_signature, signer_signature);
     }
+
+    #[test]
+    fn test_send_request_connect_timeout() {
+        let timeout_duration = Duration::from_secs(3);
+
+        // Start measuring time
+        let start_time = Instant::now();
+
+        // Attempt to send a request with a timeout
+        let result = send_request(
+            "10.255.255.1", // Non-routable IP for timeout
+            80,             // HTTP port
+            b"{}",          // Example empty JSON body
+            &Url::parse("http://10.255.255.1/").expect("Failed to parse URL"),
+            timeout_duration,
+        );
+
+        // Measure the elapsed time
+        let elapsed_time = start_time.elapsed();
+
+        // Assert that the connection attempt timed out
+        assert!(
+            result.is_err(),
+            "Expected a timeout error, but got {:?}",
+            result
+        );
+        assert_eq!(
+            result.unwrap_err().kind(),
+            std::io::ErrorKind::TimedOut,
+            "Expected a TimedOut error"
+        );
+
+        // Assert that the elapsed time is within an acceptable range
+        assert!(
+            elapsed_time >= timeout_duration,
+            "Timeout occurred too quickly"
+        );
+        assert!(
+            elapsed_time < timeout_duration + Duration::from_secs(1),
+            "Timeout took too long"
+        );
+    }
+
+    #[test]
+    fn test_send_request_timeout() {
+        // Set up a TcpListener that accepts a connection but delays response
+        let listener = TcpListener::bind("127.0.0.1:0").expect("Failed to bind test listener");
+        let addr = listener.local_addr().unwrap();
+
+        // Spawn a thread that will accept the connection and do nothing, simulating a long delay
+        thread::spawn(move || {
+            let (stream, _addr) = listener.accept().unwrap();
+            // Hold the connection open to simulate a delay
+            thread::sleep(Duration::from_secs(10));
+            drop(stream); // Close the stream
```

Comment:
Should not be necessary, `stream` is owned here and will be dropped when it goes out of scope

```suggestion
```

Reply:
If I don't include this `drop`, then we get an unused warning for `stream`.

Reply:
I guess it's not a bad idea to be explicit here, it's just a bit strange to see `drop()` explicitly called anywhere, I consider it a "code smell"

Reply:
> Ok, it seems like it still works if I just define it as `_stream`

On second thought, if you are trying to hold the stream open for a while, this might not be reliable behavior and might depend on the optimization level. I assume you're using the `debug` profile at the moment. So maybe do go with the explicit `drop()`